### DISCUSSION TITLE
[Proposal] Use SWC to transform

### DISCUSSION THREAD

**Discussion by alexzhang1030** [vue-macros/vue-macros#170] [General]
I was consider if some macros could use SWC to transorm. So I just tried.

Luckly, It looks like works great. [This is a demo](https://github.com/alexzhang1030/swc-plugin-vue-macros-define-render)

So @sxzz would you like to migrate to SWC ?

**Comment by sxzz** [comment]
Absolutely! I would like to have this to Vue Macros. Could you please provide a benchmark compare as well?

**Reply by alexzhang1030** [reply]
Yeah, I will work on this (Maybe later today?)

**Reply by sxzz** [reply]
Great! I'm looking forward.

**Reply by alexzhang1030** [reply]
I created a [repo](https://github.com/alexzhang1030/vue-macros-benchmark) to get `defineRender`'s performance benchmark. 
  For babel:  I don't know how to test vue-macros, so I just directly copied the source code in project.
  For swc: I use `@swc/core` and register plugins.
  
  Actually, I don't know if it is fair for vue-macros, cause when I use `swc`'s `transformSync`, It is about 80% slower than babel ! ( I guess sync way for node-rust binding is too expensive). So I use promises to test these two situations. 
  
  BTW, maybe using`@swc/core` is not the best practice, so swc might be faster than our thought.

**Reply by sxzz** [reply]
You didn't use await 
![33141669686673_ pic](https://user-images.githubusercontent.com/6481596/204419210-28b61cb6-d5db-4970-a215-2a4a9130eb6b.jpg)

I added that, and here is benchmark 😓
```
Running "defineRender" suite...
Progress: 100%

  swc:
    1 567 ops/s, ±7.84%    | slowest, 95.26% slower

  babel:
    33 078 ops/s, ±0.89%   | fastest

Finished 2 cases!
  Fastest: babel
  Slowest: swc
```

**Reply by alexzhang1030** [reply]
My bad ! I'm so sorry 😢 I will try to figure out why SWC is so slow
